### PR TITLE
make core_params Registry object

### PR DESCRIPTION
**Pull request by fcoulter**
Pull Request for Issue #19487 .
### Summary of Changes

This replaces previous PR

adds test for whether the core_params are a string, if they are converts them to a Registry object

Without this fix, a fatal error can occur in a content plugin that uses the parameters, as for example in the jw_disqus plugin as described in the issue tracker

### Testing Instructions

install jw_disqus plugin

create a menu item of type tags component compact tag list

add this code to components/com_tags/views/tag/view.html.php

if (is_string($itemElement->core_params))
{
$itemElement->core_params = new Registry($itemElement->core_params);
}
### Expected result

Without the added code, this causes a fatal error in Joomla 3.8.5

### Actual Result

With the added code, the page renders as it should
### Documentation Changes Required

None that I can see

### CODE REVIEW

File: components/com_tags/views/tag/view.html.php [hunk 1]
```diff
@@ -93,6 +93,11 @@ public function display($tpl = null)
 
 				// For some plugins.
 				!empty($itemElement->core_body) ? $itemElement->text = $itemElement->core_body : $itemElement->text = null;
+
```
Comment (Quy) [inline components/com_tags/views/tag/view.html.php]: Please remove tabs on line 96.

File: components/com_tags/views/tag/view.html.php [hunk 2]
```diff
@@ -93,6 +93,11 @@ public function display($tpl = null)
 
 				// For some plugins.
 				!empty($itemElement->core_body) ? $itemElement->text = $itemElement->core_body : $itemElement->text = null;
+				
+				if (is_string($itemElement->core_params))
+				{
+					$itemElement->core_params = new Registry($itemElement->core_params);
+				}
```
Comment (Quy) [inline components/com_tags/views/tag/view.html.php]: Remove trailing tabs on line 100.

File: components/com_tags/views/tag/view.html.php [hunk 3]
```diff
@@ -94,6 +94,11 @@ public function display($tpl = null)
 				// For some plugins.
 				!empty($itemElement->core_body) ? $itemElement->text = $itemElement->core_body : $itemElement->text = null;
 
+				if (is_string($itemElement->core_params))
```
Comment (csthomas) [inline components/com_tags/views/tag/view.html.php]: IMO the statement "if" is unnecessary. Take a look at line https://github.com/joomla/joomla-cms/pull/19684/files#diff-3f7bd61459465d483eb2e4ac6b359d93L123

Comment (fcoulter) [reply components/com_tags/views/tag/view.html.php]: Actually I agree with you, the treatment of the parameters seems to me to be a bit of a mess, if you look back to line 82 you have json_decode being called on a registry object

$itemElement->params = (array) json_decode($itemElement->params);

It's unclear why they are being converted to an array.

An then lower down it seems that the same params are converted back to a Registry object

	$item[0]->params = new Registry($item[0]->params);

And nothing seems to be done with them in the meantime. Its fairly confusing.

So I am think that possibly this needs more review than just my quick fix of the original issue.

Comment (fcoulter) [reply components/com_tags/views/tag/view.html.php]: Incidentally the line you mention is OK,

			$core_params = json_decode($row->core_params);

and doesn't cause any issues, and doesn't automatically mean that the core_params are expected to be a string. You can call json_encode on a Registry object, if you do it is converted to a json-formatted string first using the Registry toString method so it works. I checked it myself.

But I agree with your point all the same.

Comment (csthomas) [reply components/com_tags/views/tag/view.html.php]: I suggest to remove that line and use the Registry instance `$row->core_params`. 
I do not see a reason to encode Registry to string and decode it back to `$core_params`.

Comment (fcoulter) [reply components/com_tags/views/tag/view.html.php]: Done, also removed "if" statement.

File: components/com_tags/views/tag/view.html.php [hunk 4]
```diff
@@ -120,8 +122,7 @@ public function display($tpl = null)
 		{
 			foreach ($items as $row)
 			{
-				$core_params = json_decode($row->core_params);
-				$row->core_images = json_encode(array('image_intro' => $core_params->image, 'image_intro_alt' => $core_params->image_alt));
+				$row->core_images = json_encode(array('image_intro' => $row->core_params->get('image',''), 'image_intro_alt' => $row->core_params->get('image_alt', '')));
```
Comment (Quy) [inline components/com_tags/views/tag/view.html.php]: Add space after `'image',`

**Comment by ghost** [comment]
changed Header-Size from # to ###.

**Comment by Salgrev** [comment]
Hello,
When using tags selections, I get an error (see file). I do not use "disqus"
In order to correct this, I have to modify components/com_tags/views/tag/view.html.php in the following way : add at line 140 (joomla 3.8.12) :
Adjonction 25/02/2018 ******************************************************************************* */
if (is_string($itemElement->core_params))
{
	$itemElement->core_params = new Registry($itemElement->core_params);
}
/******************************************************************************************************** */
This lasts since february 2018... It would be nice to see it corrected !
Sincerely,
Salgrev
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19684">issues.joomla.org/tracker/joomla-cms/19684</a>.</sub>

**Comment by Salgrev** [comment]
I have tested this item :white_check_mark: successfully on 35a05928a087b9f77c6c146912638f90f7afa583<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19684">issues.joomla.org/tracker/joomla-cms/19684</a>.</sub>

**Comment by brianteeman** [comment]
@fcoulter could you resolve the conflicts please

**Comment by fcoulter** [comment]
I did it was all fine. I made this PR some 6 months ago, there were no conflicts. For whatever reason nothing further happened with it. Now there are several new releases of Joomla it is not surprising there are conflicts now.

I will see what I can do but it is not high on my list of priorities just now.

**Comment by brianteeman** [comment]
Unfortunately this happens all the time to all of us

**Comment by fcoulter** [comment]
OK, how do you approach this when it happens to you? I tried updating my fork to the latest Joomla staging but this fails because the file components/com_tags/views/tag/view.html.php  which is the subject of the original PR differs. So how do you resolve the conflicts? Not expecting a tutorial but a small clue might be helpful.

**Comment by brianteeman** [comment]
As this is a fairly simple one you can use the web editor in github by clicking on the resolve conflicts button. This will open the conflicted file in a text editor and show you where the conflict is - you can then edit it directly in the editor

**Comment by fcoulter** [comment]
OK thanks Brian.

**Comment by Salgrev** [comment]
You get a solution to this by adding on line 140 (for 3.8.12) :

/*Adjonction 25/02/2018 *******************************************************************************/
if (is_string($itemElement->core_params))
{
$itemElement->core_params = new Registry($itemElement->core_params);
}
/******************************************************************************************************* */
Enjoy
Salgrev

File: components/com_tags/views/tag/view.html.php [hunk 5]
```diff
@@ -155,11 +157,16 @@ public function display($tpl = null)
 					$itemElement->core_body = $itemElement->text;
 				}
 
+
```
Comment (Quy) [inline components/com_tags/views/tag/view.html.php]: Remove extra blank line.

Comment (fcoulter) [reply components/com_tags/views/tag/view.html.php]: Which blank line?

**Comment by fcoulter** [comment]
All checks have passed. (Let's hope) my work here is done.

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19684">issues.joomla.org/tracker/joomla-cms/19684</a>.</sub>

**Comment by cloggg** [comment]
I have tested this item :white_check_mark: successfully on 430b64fa316abe0a38790454c481decf2e9d0059<br /><br />Tested against testcase described here:
https://github.com/joomla/joomla-cms/issues/22934
Test result: positive
Thank you!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19684">issues.joomla.org/tracker/joomla-cms/19684</a>.</sub>

**Pull request merged**